src/integ_test_resources/ios/sdk/integration/cdk/cdk_integration_tests_ios/iot_stack.py

```diff
@@ -205,29 +220,126 @@ def setup_custom_authorizer(self):
+                "iot:UpdateDomainConfiguration",
+                "iot:CreateDomainConfiguration",
+                "iot:DescribeDomainConfiguration",
+                "iot:DeleteDomainConfiguration",
+            ],
+            resources=["*"],
+        )
+        provider_lambda = aws_lambda.SingletonFunction(
+            self,
+            "iot_custom_authorizer_user_pass_provider_lambda",
+            uuid="iot_custom_authorizer_user_pass_provider_lambda_20200727123737",
+            runtime=aws_lambda.Runtime.PYTHON_3_7,
+            code=aws_lambda.Code.asset("custom_resources/iot_custom_authorizer_user_pass_provider"),
+            handler="iot_custom_authorizer_user_pass_provider.on_event",
+            description="Sets up an IoT custom authorizer for user password & required domain config due to beta status",
+            environment={"custom_auth_user_pass_uuid" : Constants.custom_auth_user_pass_uuid,
+                         "custom_auth_user_pass_default_authorizer_name" : Constants.custom_auth_user_pass_default_authorizer_name,
+                         "custom_auth_user_pass_domain_configuration_name" : Constants.custom_auth_user_pass_domain_configuration_name},
+            current_version_options=aws_lambda.VersionOptions(
+                removal_policy=core.RemovalPolicy.DESTROY
+            ),
+            initial_policy=[create_authorizer_policy],
+        )
+
+        provider = custom_resources.Provider(
+            self, "iot_custom_authorizer_user_pass_provider", on_event_handler=provider_lambda
+        )
+
+        public_key = iot_custom_authorizer_key_resource.get_att(
+            "custom_authorizer_public_key"
+        ).to_string()
+
+        iot_endpoint = core.CustomResource(
+            self,
+            "iot_custom_authorizer_user_pass",
+            resource_type="Custom::IoTCustomAuthorizer",
+            service_token=provider.service_token,
+            properties={
+                "authorizer_function_arn": authorizer_function_arn,
+                "authorizer_name": custom_authorizer_name,
+                "public_key": public_key,
+                "token_key_name": token_key_name,
+            },
+        )
+        endpoint_address = iot_endpoint.get_att("BetaEndpointAddress").to_string()
+        self._parameters_to_save["iot_beta_endpoint_address"] = endpoint_address
+
+    def merge_dict(self, dict1, dict2):
+        res = {**dict1, **dict2}
+        return res
+
+    def setup_custom_authorizer_function(
+        self, unique_id, code_asset, code_handler, description, environment, region
+    ) -> str:
         """
         Sets up the authorizer Lambda, and grants 'lambda:InvokeFunction' to the service principal
         'iot.amazonaws.com'
 
         :return: the ARN of the created function
         """
+        common_vars={"RESOURCE_ARN": f"arn:aws:iot:{region}:{self.account}:*"}
+        merged_environment_vars = self.merge_dict(common_vars, environment)
```

Comment:
Consider [`update`](https://docs.python.org/3/library/stdtypes.html#dict.update)? 
```python
merged_environment_vars = {"...your initial assignment as you have above for common_vars"}
merged_environment_vars.update(environment)
```

Or reverse the order depending on which source dict you want to take precedence in the event of a key collision

Reply:
Thanks... i was looking for this!